pkg/kontainer-engine/drivers/aks/aks_driver.go

```diff
@@ -1304,7 +1304,7 @@ func (d *Driver) PostCheck(ctx context.Context, info *types.ClusterInfo) (*types
 	failureCount := 0
 
 	for {
-		info.ServiceAccountToken, err = util.GenerateServiceAccountToken(clientset)
+		info.ServiceAccountToken, err = util.GenerateServiceAccountToken(clientset, "")
```

Comment:
It's not an easy call, we could decide to only lock if `lockKey != ""` which would only lock if we knew for sure we were creating for a Cluster.

But, there are some cases (like this) where we don't necessarily have a cluster identifier, despite the fact that we're creating for a cluster.

Reply:
I think this one does have a cluster identifier. The [cluster object](https://github.com/rancher/rancher/blob/4e2703bb4f157be02dbc2f2828d99a5b1c61bb73/pkg/kontainer-engine/cluster/cluster.go#L38) that [calls the driver](https://github.com/rancher/rancher/blob/4e2703bb4f157be02dbc2f2828d99a5b1c61bb73/pkg/kontainer-engine/cluster/cluster.go#L141) has a Name field. 

That being said, any option that you can use to pass the option down (such as changing the Info object or by adding a new parameter to the drivers) may have a wider effect. The only easy option would be using the context, which I think is a bit hacky.

We could try and skip this here, as long as we are aware of the performance impact on components that use this (essentially that the post-check operations can only be done for one cluster at a time). Provisioning team (@jakefhyde ) might be able to evaluate if that trade-off is ok.

Reply:
I agree that adding it to the context is hacky.

It doesn't seem wildly unreasonable to include the `Cluster.Name` in the `ClusterInfo` (it feels slightly surprising that it's not there, given the other fields) which would fix it.

Reply:
I thought about that too - I was a bit hesitant on that one since the ClusterInfo type seems a bit unique (uses protobuf tags) and I didn't know if there was a reason for us to not touch that (i.e. if this is used in third-party drivers and could require them to be rebuilt).

That being said, the above is mostly me just not knowing the impact, if we can do it and it doesn't cause any regressions I think that's definitely the right answer.

Reply:
So far in Bullseye's experience I have never heard of users creating very large amounts of clusters via Rancher at the same time - in a lot of cases that is done via Terraform or CAPI.

Even in that case, the slowdown we are adding here is minimal - seconds at most per Cluster - where the creation of the cluster itself normally takes minutes (from my experience with EKS and AKS at least).

Therefore I agree this particular optimization can be postponed.